### PR TITLE
Add test for Spring data source configuration

### PR DESCRIPTION
**Pull request by tjwatson**

### CODE REVIEW

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

File: dev/cnf/dependabot/check_this_in_if_it_changes/pom.xml [hunk 1]
```diff
@@ -4376,6 +4376,11 @@
       <artifactId>spring-boot</artifactId>
       <version>3.2.4</version>
     </dependency>
+    <dependency>
+      <groupId>org.springframework.data</groupId>
+      <artifactId>spring-data-commons</artifactId>
+      <version>3.0.4</version>
```
Comment (anjumfatima90) [inline dev/cnf/dependabot/check_this_in_if_it_changes/pom.xml]: Is there a reason for not using version 3.2.4 here?

File: dev/io.openliberty.checkpoint_fat_springboot_data/fat/src/io/openliberty/checkpoint/fat/springboot/data/SpringBootDeployDataTests.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            case testSpringBootAfterAppStart -> CheckpointPhase.AFTER_APP_START;
+            case testSpringBootBeforeAppStart -> CheckpointPhase.BEFORE_APP_START;
+            default -> throw new IllegalArgumentException("Unexpected value: " + getTestMethodSimpleName(testName));
+        };
+        setUp(server, false, testPhase, testMethod.toString());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        server.stopServer();
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_springboot_data/fat/src/io/openliberty/checkpoint/fat/springboot/data/SpringBootDeployDataTests.java:60]: Is clearing out the server.env not necessary here based on the spring jpa properties properties set in bootstrap.properties?

File: dev/io.openliberty.checkpoint_fat_springboot_data/publish/servers/checkpointSpringBootData/server.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+<server>
+	<applicationManager startTimeout="90s"/>
+	<featureManager>
+		<feature>springBoot-3.0</feature>
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_springboot_data/publish/servers/checkpointSpringBootData/server.xml:16]: I guess it wouldn't matter much but the war test would not require the springBoot-3.0 feature.

File: dev/io.openliberty.checkpoint_fat_springboot_data/build.gradle [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_springboot_data/build.gradle]: Copyright year 2024?

**Review by anjumfatima90 — COMMENTED** [review]
Left few comments, otherwise everything looks good. Thanks!

File: dev/cnf/dependabot/check_this_in_if_it_changes/pom.xml (hunk 1 above)
Comment (tjwatson) [reply dev/cnf/dependabot/check_this_in_if_it_changes/pom.xml]: I can update the `oss_dependencies.maven` file to use 3.2.4.  That is what the `build.gradle` for the test application ends up pulling in based on  `org.springframework.boot:spring-boot-starter-data-jpa` used.  `org.springframework.data` is a separate group ID from `org.springframework.boot` so it is not clear that this artifact keeps its version in sync with spring boot.

File: dev/io.openliberty.checkpoint_fat_springboot_data/publish/servers/checkpointSpringBootData/server.xml (hunk 3 above)
Comment (tjwatson) [reply dev/io.openliberty.checkpoint_fat_springboot_data/publish/servers/checkpointSpringBootData/server.xml:16]: It does not harm and allows me to keep a single server.xml config for the test.

**Comment by tjwatson** [comment]
#build

**Review by anjumfatima90 — DISMISSED** [review]

**Comment by tjwatson** [comment]
#libby

**Comment by tjwatson** [comment]
#libby

**Review by anjumfatima90 — APPROVED** [review]

**Pull request merged**